test/test_cpp_extensions.py

```diff
@@ -106,6 +106,83 @@ def test_optional(self):
         has_value = cpp_extension.function_taking_optional(None)
         self.assertFalse(has_value)
 
+    def test_inline_jit_compile_extension(self):
+        cpp_source1 = '''
+        #include <torch/torch.h>
+        at::Tensor sin_add(at::Tensor x, at::Tensor y) {
+          return x.sin() + y.sin();
+        }
+        '''
+
+        cpp_source2 = '''
+        #include <torch/torch.h>
+        at::Tensor sin_add(at::Tensor x, at::Tensor y);
+        PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
+          m.def("sin_add", &sin_add, "sin(x) + sin(y)");
+        }
+        '''
+
+        module = torch.utils.cpp_extension.load_inline(
+            name='inline_jit_extension',
+            cpp_sources=[cpp_source1, cpp_source2],
+            verbose=True)
+
+        x = torch.randn(4, 4)
+        y = torch.randn(4, 4)
+
+        z = module.sin_add(x, y)
+        self.assertEqual(z, x.sin() + y.sin())
+
+    @unittest.skipIf(not TEST_CUDA, "CUDA not found")
+    def test_inline_jit_compile_cuda_extension(self):
+        cpp_source = '''
+        #include <torch/torch.h>
+
+        void cos_add_cuda(const double* x, const double* y, double* output, int size);
+        at::Tensor cos_add(at::Tensor x, at::Tensor y) {
+          auto output = at::zeros_like(x);
+          cos_add_cuda(x.data<double>(), y.data<double>(), output.data<double>(), output.numel());
+          return output;
+        }
+        PYBIND11_MODULE(TORCH_EXTENSION_NAME, m) {
+          m.def("cos_add", &cos_add, "cos(x) + cos(y)");
+        }
+        '''
+
+        cuda_source = '''
+        #include <cuda.h>
+        #include <cuda_runtime.h>
```

Comment:
Do we actually need those cuda includes? I think ATen already includes them, right?

Reply:
No we never include `cuda.h` in headers, only in implementation files

Reply:
Oh, I see. I didn't need to add those because I included in my files `<THC/THC.h>`, which [includes `cuda.h` and `cuda_runtime.h`](https://github.com/pytorch/pytorch/blob/master/aten/src/THC/THCGeneral.h.in#L12-L13).